test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ControllerActionInvokerTest.cs

```diff
@@ -364,26 +364,40 @@ public class ControllerActionInvokerTest
         public async Task InvokeAction_InvokesAsyncAuthorizationFilter_ShortCircuit()
         {
             // Arrange
-            var challenge = new Mock<IActionResult>(MockBehavior.Loose).Object;
+            var challenge = new Mock<IActionResult>(MockBehavior.Strict);
+            challenge
+                .Setup(r => r.ExecuteResultAsync(It.IsAny<ActionContext>()))
+                .Returns(Task.FromResult(true))
+                .Verifiable();
 
             var filter1 = new Mock<IAsyncAuthorizationFilter>(MockBehavior.Strict);
             filter1
                 .Setup(f => f.OnAuthorizationAsync(It.IsAny<AuthorizationFilterContext>()))
                 .Returns<AuthorizationFilterContext>((context) =>
                 {
-                    context.Result = challenge;
                     return Task.FromResult(true);
                 })
                 .Verifiable();
 
-            var filter2 = new Mock<IAuthorizationFilter>(MockBehavior.Strict);
+            var filter2 = new Mock<IAsyncAuthorizationFilter>(MockBehavior.Strict);
+            filter2
+                .Setup(f => f.OnAuthorizationAsync(It.IsAny<AuthorizationFilterContext>()))
+                .Returns<AuthorizationFilterContext>((context) =>
+                {
+                    context.Result = challenge.Object;
+                    return Task.FromResult(true);
+                })
+                .Verifiable();
```

Comment:
If this filter is being short-circuited, then you don't need to set any expectations on it.